csharp_dotnetcore/13.Basic-Bot-Template/Dialogs/Greeting/GreetingDialog.cs

```diff
@@ -0,0 +1,169 @@
+        /// <param name="logger">The <see cref="ILogger"/> that enables logging.</param>
+        public GreetingDialog(
+                    BotServices botServices,
+                    IStatePropertyAccessor<DialogState> dialogStateAccessor,
+                    IStatePropertyAccessor<GreetingState> greetingStateAccessor,
+                    ILogger logger)
+            : base(botServices, nameof(GreetingDialog), logger)
+        {
+            DialogStateAccessor = dialogStateAccessor ?? throw new ArgumentNullException($"Missing parameter. {nameof(dialogStateAccessor)} is required.");
+            GreetingStateAccessor = greetingStateAccessor ?? throw new ArgumentNullException($"Missing parameter. {nameof(greetingStateAccessor)} is required.");
+
+            // Add control flow dialogs
+            var waterfallSteps = new WaterfallStep[]
+            {
+                InitializeStateStepAsync,
+                PromptForNameStepAsync,
+                PromptForCityStepAsync,
+                DisplayGreetingStateStepAsync,
+            };
+            AddDialog(new WaterfallDialog(ProfileDialog, waterfallSteps));
+            AddDialog(new NamePrompt(nameof(NamePrompt)));
+            AddDialog(new CityPrompt(nameof(CityPrompt)));
+        }
+
+        public IStatePropertyAccessor<DialogState> DialogStateAccessor { get; }
+
+        public IStatePropertyAccessor<GreetingState> GreetingStateAccessor { get; }
+
+        private BotServices BotServices { get; }
+
+        private async Task<DialogTurnResult> InitializeStateStepAsync(
+                                                          DialogContext dc,
+                                                          WaterfallStepContext stepContext,
+                                                          CancellationToken cancellationToken)
+        {
+            var greetingState = await GreetingStateAccessor.GetAsync(dc.Context, () => new GreetingState());
+
+            stepContext.Values[NameValue] = greetingState.Name;
+            stepContext.Values[CityValue] = greetingState.City;
+
+            return await stepContext.NextAsync();
+        }
+
+        private async Task<DialogTurnResult> PromptForNameStepAsync(
+                                                DialogContext dc,
+                                                WaterfallStepContext stepContext,
+                                                CancellationToken cancellationToken)
+        {
+            // Prompt for name if missing
+            if (string.IsNullOrWhiteSpace((string)stepContext.Values[NameValue]))
+            {
+                var options = new PromptOptions
+                {
+                    Prompt = new Activity
+                    {
+                        Type = ActivityTypes.Message,
+                        Text = "What is your name ?",
+                    },
+                };
+
+                return await dc.PromptAsync(nameof(NamePrompt), options).ConfigureAwait(false);
```

Comment:
can't this simply be:
await dc.PromptAsync(nameof(NamePrompt), "what is your name?", options) ?

Or am I missing something?

Reply:
You must have overheard us, that's checked in!